projects/apple/tomviz.bundle.cmake

```diff
@@ -9,52 +9,103 @@ set(CPACK_DMG_DS_STORE_SETUP_SCRIPT
-    # be 'fixed' using otool.
-    # qt/plugins is needed as QtPrintSupport.framework directly links to the printsupport plugin
-    execute_process(
-       COMMAND ${CMAKE_CURRENT_LIST_DIR}/fixup_bundle.py
-               --exe \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/tomviz.app\"
-               --search \"${install_location}/lib\"
-               --search \"${Qt5_DIR}/../../../lib\"
-               --search \"${Qt5_DIR}/../../../plugins\"
-               --plugins \"${Qt5_DIR}/../../../plugins\")
-   "
-   COMPONENT superbuild)
+set(superbuild_install_location ${install_location})
+include(SuperbuildInstallMacros)
+
+superbuild_apple_create_app(
+  "\${CMAKE_INSTALL_PREFIX}"
+  "tomviz.app"
+  "${superbuild_install_location}/Applications/tomviz.app/Contents/MacOS/tomviz"
+  CLEAN
+  SEARCH_DIRECTORIES "${superbuild_install_location}/lib" "${Qt5_DIR}/../../../lib" "${Qt5_DIR}/../../../plugins"
+  INCLUDE_REGEXES ".*/libgfortran" ".*/libquadmath")
+
+install(
+  FILES       "${superbuild_install_location}/Applications/tomviz.app/Contents/Resources/tomviz.icns"
+  DESTINATION "tomviz.app/Contents/Resources"
+  COMPONENT   superbuild)
+install(
+  FILES       "${superbuild_install_location}/Applications/tomviz.app/Contents/Info.plist"
+  DESTINATION "tomviz.app/Contents"
+  COMPONENT   superbuild)
+
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/qt.conf" "[Paths]\nPlugins = Plugins\n")
+install(
+  FILES       "${CMAKE_CURRENT_BINARY_DIR}/qt.conf"
+  DESTINATION "tomviz.app/Contents/Resources"
+  COMPONENT   superbuild)
+
+install(
+  DIRECTORY "${superbuild_install_location}/Applications/tomviz.app/Contents/share"
+  DESTINATION "tomviz.app/Contents"
+  COMPONENT superbuild)
+
+# Install the core python modules we got when building python itself (this skips site-packages
+# since that is covered later.
+install(
+  DIRECTORY "${superbuild_install_location}/lib/python3.6/"
+  DESTINATION "tomviz.app/Contents/Python"
+  COMPONENT superbuild
+  PATTERN "site-packages/*" EXCLUDE
+  PATTERN "*.pyc" EXCLUDE)
+
+# ITK doesn't set up rpaths in its python modules even though it has @rpath in where to find
+# the libraries.  So we need to strip the rpaths (until we fix ITK to put rpaths in the python
+# modules).
+install(CODE
+  "execute_process(
+    COMMAND ${CMAKE_CURRENT_LIST_DIR}/fixup_itk.py \"${superbuild_install_location}/lib/itk\")"
+  COMPONENT superbuild)
+
+# ITK dumps a bunch of stuff in the root site_packages directory.  I've factored this list out
+# into its own file since it is huge.  This will need changing if we enable more of ITK.
+# This include brings in one variable: itk_python_modules which is a list of all the root ITK
+# python modules.
+include(itk_list)
+
+superbuild_apple_install_python(
+  "\${CMAKE_INSTALL_PREFIX}"
+  "tomviz.app"
+  MODULES tomviz
+          paraview
+          pygments
+	  vtk
+	  vtkmodules
+          ${itk_python_modules}
+          pyfftw
+          numpy
+          scipy
+          pyfftw
```

Comment:
`paraview` and `vtk` are in this list. Ideally this would just have `itk` instead of `${itk_python_modules}` but we can't do that. It looks through the list of directories (`MODULE_DIRECTORIES`) below and finds the modules requested and recursively installs them. And it looks for dependencies they don't provide themselves in the locations given in `SEARCH_DIRECTORIES`.

Also, apparently I have `pyfftw` in the list twice... oops.

Reply:
There is some odd indentation in there too, so what you are saying is that the ITK directory structure is odd/doesn't fit the pattern used in other Python modules, necessitating the hand-curated list? I wonder if @thewtex knows of a way to get that list without hand-curating it - is it in a config file for example?

Reply:
There is nothing non-standard with the ITK Python modules -- there is just a lot of good stuff ;-).

To avoid a hard list of the modules, you could glob `site-packages/itk/Configuration`.

Reply:
I would like to avoid hard coded lists, so any option you can give us to get the information we need for packaging would be useful. I don't want to have to update this list, or have subtle bugs because we didn't add a new module when we bump ITK next. Thanks for the info, @mathturtle can you check out whether globbing as suggested will work for us.

Reply:
@thewtex The difference is that with `numpy` I can install the `site-packages/numpy` folder and get all the numpy-related stuff.  With ITK, I need the `itk` folder plus `itkLazy.py`, `itkBase.py`, plus what looks like one other file per module which are all in the root `site-packages` folder.  I'll try the globbing approach now and update this branch if it works.

PS: If you could make the RPATH in the ITK python modules point to the other ITK libraries so that an install tree works without setting `DYLD_LIBRARY_PATH` that would be really nice.  Right now I have a script that strips the `@rpath` out of the dependent libraries and resets them to full paths since this isn't working.  The rpath in the python c++ modules needs to be something like `@loader_path/../..` (look at ParaView's python modules for an example where this works).  Note: tomviz still uses ITK 4.12, so this may be fixed in more recent ones?